### PR TITLE
Review request for Microsoft.DBforPostgreSQL to add version preview/2024-03-01-privatepreview

### PR DESCRIPTION
**Pull request by nasc17**
# ARM (Control Plane) API Specification Update Pull Request 

Changes included in this PR:

1. New preview API version added: 2024-03-01-privatepreview
2. Add support to get, create, and delete requests for customer on-demand backups (Backups.json)
3. Add support to update administratorLogin (FlexibleServers.json)

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

spec_pr_review_workflow_diagram

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [X ] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [X ] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [ X] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information
Changes in current PR:
1.	New preview API version added: 2024-03-01-privatepreview
2.	Added PUT api to Create Backup and DELETE backup in Backups.json.
3. Added new property "administratorLogin" for update of flexible server in FlexibleServers.json

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories")
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

### CODE REVIEW

**Review by ambrahma — APPROVED** [review]

**Comment by razvanbadea-msft** [comment]
Please fill out the Purpose of this PR and Due diligence checklist

**Comment by ambrahma** [comment]
> Please fill out the Purpose of this PR and Due diligence checklist

Updated

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2024-03-01-preview/TuningOptions.json
Comment (razvanbadea-msft) [inline specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2024-03-01-preview/TuningOptions.json:292]: You should add in the description the format

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2024-03-01-preview/TuningOptions.json
Comment (razvanbadea-msft) [inline specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2024-03-01-preview/TuningOptions.json:297]: You should add in the description the format

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2024-03-01-preview/TuningOptions.json
Comment (razvanbadea-msft) [inline specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2024-03-01-preview/TuningOptions.json:187]: Could the type of this be converted to a enum?

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2024-03-01-preview/TuningOptions.json
Comment (razvanbadea-msft) [inline specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2024-03-01-preview/TuningOptions.json:219]: could this be converted to an enum?

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2024-03-01-preview/TuningOptions.json
Comment (razvanbadea-msft) [inline specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2024-03-01-preview/TuningOptions.json:233]: shouldn't here be an array? or at least add in the description how are the values separated

Comment (ambrahma) [reply specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2024-03-01-preview/TuningOptions.json:233]: Updated description to include will be semi-colon separated

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2024-03-01-preview/TuningOptions.json
Comment (ambrahma) [reply specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2024-03-01-preview/TuningOptions.json]: UPdated description to include format

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2024-03-01-preview/TuningOptions.json
Comment (ambrahma) [reply specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2024-03-01-preview/TuningOptions.json:187]: Updated description to include format

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2024-03-01-preview/TuningOptions.json
Comment (ambrahma) [reply specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2024-03-01-preview/TuningOptions.json]: Updated description to include format

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2024-03-01-preview/TuningOptions.json
Comment (ambrahma) [reply specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2024-03-01-preview/TuningOptions.json:187]: THis is what I hear from backend team:
Here at this moment we can have 2 values:
MB
Percentage
Nevertheless we would prefer not to have enums, because we will be constrained. Ex, for now we went with MB, but we might choose another unit. So ideally at least for prewiew would be great to have it as a string to give us more flexibility

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2024-03-01-preview/TuningOptions.json
Comment (ambrahma) [reply specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2024-03-01-preview/TuningOptions.json:219]: For now only btree is supported and might be so for quite a while, albeit Postgres has many other index types. I don't think worths adding an enum with 1 single element

**Comment by ms-bogdan-rotaru** [comment]
Please check the errors reported by the required checks.

**Comment by ambrahma** [comment]
> Please check the errors reported by the required checks.

.NET SDK errors popped up after latest merge with master. However, these errors do not seem related to current PR as we havent changed anything in FlexibleServers.json (This is the file whose entities are being reflected in the error).

Also the go-sdk error is not something which I understand here.

I think we can ignore Avocado error here. This PR needs ARM approval so that we can go ahead and check-in ARM manifest to include new auto-migration APIs added to Migrations.json.

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2024-03-01-preview/Capabilities.json
Comment (ambrahma) [inline specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2024-03-01-preview/Capabilities.json:274]: check if this is required even for ServerCapabilities API

**Review by ambrahma — APPROVED** [review]

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2024-03-01-preview/Capabilities.json
Comment (nasc17) [reply specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2024-03-01-preview/Capabilities.json:274]: Validated and updated examples

**Pull request merged**
